### DISCUSSION TITLE
Unable to connect to MySQL server, the connection request does not contain the (configured) password.

### DISCUSSION THREAD

**Discussion by Stipriaan** [mkleehammer/pyodbc#1395] [Q&A]
When trying to connect to the MySQL server the following error is returned :
pyodbc.Error: ('HY000', "[HY000] [MySQL][ODBC 8.0(a) Driver]Access denied for user 'stip'@'192.168.188.254' (using password: NO) (1045) (SQLDriverConnect); [HY000] [MySQL][ODBC 8.0(a) Driver]Access denied for user 'stip'@'192.168.188.254' (using password: NO) (1045)")

The password is contained in the connection string.
When I make a packet capture of the connection the password is not in the connection request that is sent to the MySQL server.
The connection is made to the correct server and the username and database name are visible in the request.
Why is the password missing in the connection request ?
Do I need to configure some additional parameters ?
pyodbc version is 5.2.0
python version is 3.10.4

**Comment by gordthompson** [comment]
By chance does your password begin with a semicolon (`;`)?

**Reply by Stipriaan** [reply]
No special characters in the password, just lowercase and uppercase letters and digits.

**Reply by gordthompson** [reply]
What does your connection string look like?

**Reply by Stipriaan** [reply]
connectionString = f'DRIVER={{MySQL ODBC 8.0 ANSI Driver}};SERVER={SERVER};DATABASE={DATABASE};UID={USERNAME};PWD={PASSWORD}'
If I print the connectionString it shows me the correct variables.
If I look at the packet capture of the connection to the MySQL server the database name and user id are included but the password isn't.

**Reply by gordthompson** [reply]
I am unable to reproduce your issue. When I use pyodbc 5.2.0 to (successfully) connect to my old MySQL 5.7.32 server using …

```python
    connection_string = (
        'DRIVER=MySQL ODBC 8.2 ANSI Driver;'
        'SERVER=192.168.0.199;'
        'PORT=3306;'
        'DATABASE=test;'
        'SSLMODE=DISABLED;'
        'UID=stip;'
        'PWD=abc123;'
    )
```

… Wireshark shows me

```
Frame 5: 348 bytes on wire (2784 bits), 348 bytes captured (2784 bits) on interface \Device\NPF_{CCEDF5E2-CE7C-43BF-93DD-F784A668AB04}, id 0
Ethernet II, Src: Universa_34:c0:d3 (40:2c:f4:34:c0:d3), Dst: Dell_75:83:c1 (00:21:70:75:83:c1)
Internet Protocol Version 4, Src: 192.168.0.179, Dst: 192.168.0.199
Transmission Control Protocol, Src Port: 62558, Dst Port: 3306, Seq: 1, Ack: 79, Len: 294
MySQL Protocol
    Packet Length: 290
    Packet Number: 1
    Login Request
        Client Capabilities: 0xa28d
        Extended Client Capabilities: 0x19be
        MAX Packet: 1073741824
        Charset: latin1 COLLATE latin1_swedish_ci (8)
        Unused: 0000000000000000000000000000000000000000000000
        Username: stip
        Password: 870ac33d039c46e0efb0cffe1f05b741293ec6d6
        Schema: test
        Client Auth Plugin: mysql_native_password
        Prefix: 204
        Length: 204
        Connection Attributes
```

**Reply by Stipriaan** [reply]
This is what WhireShark shows me :

```
Frame 5: 105 bytes on wire (840 bits), 105 bytes captured (840 bits) on interface \Device\NPF_{D267EF3B-04D8-4E7A-A8EA-FA2CD189CA87}, id 0
Ethernet II, Src: HewlettPacka_a1:c2:d3 (78:e3:b5:a1:c2:d3), Dst: JuniperNetwo_07:a5:cb (2c:6b:f5:07:a5:cb)
Internet Protocol Version 4, Src: 172.27.4.71, Dst: 192.168.188.11
Transmission Control Protocol, Src Port: 15946, Dst Port: 3306, Seq: 1, Ack: 57, Len: 51
MySQL Protocol
    Packet Length: 47
    Packet Number: 1
    Login Request
        Client Capabilities: 0xa28d
        Extended Client Capabilities: 0x19be
        MAX Packet: 1073741824
        Collation: latin1 COLLATE latin1_swedish_ci (8)
        Unused: 0000000000000000000000000000000000000000000000
        Username: stip
        Schema: dbmn2425
        Client Auth Plugin: 

```

**Reply by gordthompson** [reply]
I wonder why "Client Auth Plugin:" is empty. If you can connect some other way (e.g., MySQL Workbench, or similar) do you get anything back from

```sql
show variables like '%default_auth%'
```

Also, for diagnostic purposes, can you try connecting from Python using [PyMySQL](https://pypi.org/project/PyMySQL/)?
